frappe/custom/doctype/property_setter/property_setter.py

```diff
@@ -3,16 +3,32 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
 
 from frappe.model.document import Document
 
+not_allowed_fieldtype_change = ['naming_series']
+
 class PropertySetter(Document):
 	def autoname(self):
 		self.name = self.doc_type + "-" \
 			+ (self.field_name and (self.field_name + "-")  or "") \
 			+ self.property
 
 	def validate(self):
+		self.validate_fieldtype_change()
+		self.delete_property_setter()
+
+		# clear cache
+		frappe.clear_cache(doctype = self.doc_type)
+
+	def validate_fieldtype_change(self):
+		if self.field_name in not_allowed_fieldtype_change and \
+			self.property == 'fieldtype':
+			frappe.throw(_("""Fieldtype cannot be changed
```

Comment:
Translatable strings should not be multiline

Better message "Field type cannot be changed for {0}"